content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1389 @@
+                        <add input="{HTTP_HOST}" pattern="^jenkins\.acme\.example$" />
+                    </conditions>
+                </rule>
+            </rules>
+        </rewrite>
+        <security>
+            <requestFiltering allowDoubleEscaping="true" />
+        </security>
+    </system.webServer>
+</configuration>
+----
+
+
+===== Complain about it still not working
+
+Leave a comment or complain to me over at twitter
+(@Darsstar) +
+
+=== Running Jenkins behind Squid
+
+In situations where you want a user friendly url to access Jenkins (Not
+port 8080), it makes sense to have Jenkins runs behind Squid, so that
+you can access Jenkins on port 80 or 443. 
+This section discusses some of the approaches for doing this.
+
+==== Squid 2.6
+
+Using Squid 2.6 (default on Centos 5.3):
+
+[source]
+----
+acl all src 0.0.0.0/0.0.0.0
+acl localhost src 127.0.0.1/255.255.255.255
+acl manager proto cache_object
+acl to_localhost dst 127.0.0.0/8
+acl valid_dst dstdomain .YOUR_DOMAIN ci
+
+
+cache_replacement_policy heap LFUDA
+memory_replacement_policy heap GDSF
+
+cache_dir ufs /var/spool/squid 512 16 256
+cache_mem 512 MB
+maximum_object_size 12000 KB
+
+## http --> https redirect
+## don't forget to update "Jenkins URL" on https://ci.YOUR_DOMAIN/configure
+#acl httpPort myport 80
+#http_access deny httpPort
+#deny_info https://ci.YOUR_DOMAIN/ httpPort
+
+
+cache_peer localhost parent 8080 0 originserver name=myAccel
+coredump_dir /var/spool/squid
+hierarchy_stoplist cgi-bin
+http_access allow localhost
+http_access allow manager localhost
+http_access allow valid_dst
+http_access deny all
+http_access deny manager
+
+## mkdir /etc/squid/ssl/ && cd /etc/squid/ssl/ # to generate your self-signed certificate
+## openssl genrsa -out jenkins.key 1024
+## openssl req -new -key jenkins.key -x509 -out jenkins.crt -days 999
+http_port 80 vhost
+#https_port 443 cert=/etc/squid/ssl/jenkins.crt key=/etc/squid/ssl/jenkins.key vhost
+
+http_reply_access allow all
+icp_access allow all
+
+refresh_pattern -i \.jp(e?g|gif|png|ico)   300  20%  600 override-expire
+
+
+logformat combined %>a %ui %un \[%tl\] "%rm %ru HTTP/%rv" %Hs %<st "%{Referer}>h" "%{User-Agent}>h" %Ss:%Sh
+strip_query_terms off
+access_log /var/log/squid/access.log combined
+
+visible_hostname ci.YOUR_DOMAIN
+----
+
+This assumes that you run Jenkins on localhost port 8080. 
+But you can have it on an other server / different port 
+(adjust line starting with cache_peer) +
+
+Of course replace  YOUR_DOMAIN with your domain. +
+
+===== With ssl
+
+Remove one level of comment
+
+[source]
+----
+ sed s/^#// /etc/squid/squid.conf
+----
+
+/!\ If you use the swarm client plugin , the nodes will complain about :
+
+[source]
+----
+Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```

Comment:
Split the long line for readability and reduced impact on the rest of the text on the page.
```suggestion
Caused by: sun.security.validator.ValidatorException:
    PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException:
        unable to find valid certification path to requested target
```